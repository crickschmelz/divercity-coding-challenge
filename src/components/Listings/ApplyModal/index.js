import React, { useState } from "react";
import {
  Modal,
  Button,
  Form,
  Input,
  ExitButton,
  SuccessModal,
  RegisterHeader
} from "./style";
import { applyToJob } from "../../utils";

const ApplyModal = (props) => {
  const { closeApplication } = props;
  const [motivation, setMotivation] = useState("");
  const [cover_letter, setCoverLetter] = useState("");
  const [showMessage, setShowMessage] = useState(false);

  const openMessage = (e) => {
    setShowMessage(true);
    setTimeout(() => {
      setShowMessage(false);
      closeApplication();
    }, 2000);
    e.currentTarget.reset();
  };

  const apply = (e) => {
    e.preventDefault();
    const token = localStorage.getItem("token");
    applyToJob("https://divercity-test.herokuapp.com/jobs/2/apply", token, {
      motivation,
      cover_letter
    }).then((data) => {
      return data;
    });
    openMessage(e);
  };

  return (
    <Modal>
      <ExitButton onClick={closeApplication}>x</ExitButton>
      <RegisterHeader>Apply</RegisterHeader>
      {showMessage && (
        <SuccessModal>
          <p>Success! You've applied to this job.</p>
        </SuccessModal>
      )}
      <Form onSubmit={apply} enctype="application/x-www-form-urlencoded">
        <Input
          type="textarea"
          required
          placeholder="Why do you want the job?"
          onChange={(event) => setMotivation(event.target.value)}
        ></Input>
        <Input
          type="textarea"
          required
          placeholder="Anything else to add? (i.e. cover letter)"
          onChange={(event) => setCoverLetter(event.target.value)}
        ></Input>
        <Button type="submit">Submit</Button>
      </Form>
    </Modal>
  );
};

export default ApplyModal;
