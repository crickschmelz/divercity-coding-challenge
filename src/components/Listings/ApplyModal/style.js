import styled from "styled-components";

export const RegisterHeader = styled.h3`
  color: #333241;
  padding-top: 5%;
  text-align: center;
`;

export const SuccessModal = styled.div`
  position: absolute;
  z-index: 1;
  height: 25%;
  width: 100%;
  color: rgb(37, 55, 85);
  font-weight: 800;
  background-color: #ffffff;
  opacity: 0.95;
  text-align: center;
`;

export const ExitButton = styled.button`
  color: #ffffff;
  background-color: transparent;
  border: none;
  font-weight: bold;
  font-size: 24px;
  padding: 1%;
  margin: 1% 3%;
  cursor: pointer;
  float: right;
`;

export const Button = styled.button`
  color: #ffffff;
  background-color: #3197e4;
  border: none;
  font-weight: 800;
  font-size: 14px;
  width: 100px;
  padding: 10px;
  margin: 20px 0 20px 100px;
  border-radius: 30px;
  cursor: pointer;
`;

export const Form = styled.form`
  display: grid;
  justify-content: center;
`;

export const Modal = styled.div`
  position: absolute;
  z-index: 1;
  margin: 0;
  height: 350%;
  width: 100%;
  background-color: #d8d8d8;
  border: 1px solid #bbb;
  opacity: 0.95;
  border-radius: 4px;
`;

export const Input = styled.textarea`
  display: block;
  padding: 2%;
  margin-top: 5%;
  background-color: #ffffff;
  border: none;
  width: 300px;
  height: 100px;
  border-radius: 3px;
`;
