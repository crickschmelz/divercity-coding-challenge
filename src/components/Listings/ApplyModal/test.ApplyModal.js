import React from "react";
import ApplyModal from "./ApplyModal";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<ApplyModal />).toJSON();
  expect(tree).toMatchSnapshot();
});
