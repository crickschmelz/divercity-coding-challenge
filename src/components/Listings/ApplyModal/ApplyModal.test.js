import React from "react";
import ApplyModal from "./";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<ApplyModal />).toJSON();
  expect(tree).toMatchSnapshot();
});
