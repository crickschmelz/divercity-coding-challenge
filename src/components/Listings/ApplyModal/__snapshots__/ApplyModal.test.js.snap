// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxgMl hdlxbY"
>
  <button
    className="sc-AxiKw kbLOIl"
  >
    x
  </button>
  <h3
    className="sc-AxjAm irvukv"
  >
    Apply
  </h3>
  <form
    className="sc-AxhUy bBqmEe"
    onSubmit={[Function]}
  >
    <textarea
      className="sc-AxheI dvQBCG"
      onChange={[Function]}
      placeholder="Why do you want the job?"
      required={true}
      type="textarea"
    />
    <textarea
      className="sc-AxheI dvQBCG"
      onChange={[Function]}
      placeholder="Anything else to add? (i.e. cover letter)"
      required={true}
      type="textarea"
    />
    <button
      className="sc-AxhCb fwyxFT"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
